File: onlineCourses/ae/02_medium/sort_stack.java
import java.util.*;

class Program {

  public ArrayList<Integer> sortStack(ArrayList<Integer> stack) {
    // Write your code here.
    if(stack.size() == 0) {
      return stack;
    }

    int top = stack.remove(stack.size()-1);

    sortStack(stack);

    insertInSortedOrder(stack, top);

    return stack;
  }

  public void insertInSortedOrder(ArrayList<Integer> stack, int value) {
    if(stack.size()==0 || stack.get(stack.size()-1)<=value) {
      stack.add(value);
      return;
    }

    int top = stack.remove(stack.size()-1);

    insertInSortedOrder(stack, value);

    stack.add(top);
  }
}

File: onlineCourses/ae/02_medium/sort_stack.cpp
#include <vector>
using namespace std;

void insertInSortedOrder(vector<int> &stack, int value) {
	if(stack.size()==0 || stack.back()<=value) {
		stack.push_back(value);
		return;
	}

	int top = stack.back();
	stack.pop_back();

	insertInSortedOrder(stack, value);

	stack.push_back(top);
}

vector<int> sortStack(vector<int> &stack) {
  // Write your code here.
	if(stack.size() == 0) {
		return stack;
	}

	int top = stack.back();
	stack.pop_back();

	sortStack(stack);
	insertInSortedOrder(stack, top);

	return stack;
}
